ereuse_devicehub/inventory/views.py

```diff
@@ -0,0 +1,105 @@
+import flask
+from flask.views import View
+from flask import Blueprint, url_for
+from flask_login import login_required, current_user
+
+from ereuse_devicehub.resources.lot.models import Lot
+from ereuse_devicehub.resources.device.models import Device
+from ereuse_devicehub.inventory.forms import LotDeviceForm, LotForm
+
+devices = Blueprint('inventory.devices', __name__, url_prefix='/inventory')
+
+
+class DeviceListView(View):
+    decorators = [login_required]
+    template_name = 'inventory/device_list.html'
+
+    def dispatch_request(self, id=None):
+        # TODO @cayop adding filter
+        filter_types = ['Desktop', 'Laptop', 'Server']
+        lots = Lot.query.filter(Lot.owner_id == current_user.id)
+        lot = None
+        if id:
+            lot = lots.filter(Lot.id == id).one()
+            devices = [dev for dev in lot.devices if dev.type in filter_types]
+        else:
+            devices = Device.query.filter(
+                Device.owner_id == current_user.id).filter(
+                Device.type.in_(filter_types)).filter(Device.lots == None)
```

Comment:
Why is filter applied on a different way in if than in else statement? I mean, on if filter is performed using python but on else is performed by SQL.  It will be better to do by SQL on both and merge common code.